### PR TITLE
Added ruby properties pass

### PR DESCRIPTION
**Pull request by ankit-privado**
Added `RubyPropertyLinkerPass.scala` and `RubyDBConfigTagger.scala` passes.

### CODE REVIEW

File: src/main/scala/ai/privado/languageEngine/ruby/passes/config/RubyPropertyLinkerPass.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import io.shiftleft.codepropertygraph.generated.nodes.{AstNode, JavaProperty}
+import org.slf4j.LoggerFactory
+import io.shiftleft.semanticcpg.language.*
+
+class RubyPropertyLinkerPass(cpg: Cpg) extends PrivadoParallelCpgPass[JavaProperty](cpg) {
+
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  val dbConnectionRegex =
+    "^(db|database|jdbc|mysql|postgres|oracle|sqlserver)_(connection_)?(host|port|name|user|password|uri|driver|ssl|pool_size|timeout|connection_string)$"
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/languageEngine/ruby/passes/config/RubyPropertyLinkerPass.scala:15]: Pick up the regex from privado config, don't hard code here

Comment (ankit-privado) [reply src/main/scala/ai/privado/languageEngine/ruby/passes/config/RubyPropertyLinkerPass.scala]: This regex is not needed here. i'm removing it.

**Review by karan-batavia — CHANGES_REQUESTED** [review]

File: src/main/scala/ai/privado/tagger/config/RubyDBConfigTagger.scala [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  private def parsePropForDynamoDB(dbUrl: JavaProperty): Unit = {
+    // Example dynamodb url :- https://dynamodb.<region>.amazonaws.com/<table-name>
+
+    try {
+      val dbVendor   = "dynamodb"
+      val dbLocation = dbUrl.value.split("\\.")(1)
+      val dbName     = dbUrl.value.split("/").last
+      val configFile = dbUrl.sourceFileOut.toList.head.name
+
+      DatabaseDetailsCache.addDatabaseDetails(
```
[the remark this answers is not in the document]
Comment (ankit-privado) [reply src/main/scala/ai/privado/tagger/config/RubyDBConfigTagger.scala:43]: yes sure, will add priority mechanism.

[the remark this answers is not in the document]
Comment (ankit-privado) [reply src/main/scala/ai/privado/tagger/config/RubyDBConfigTagger.scala]: Done.

File: src/main/scala/ai/privado/utility/Utilities.scala [hunk 3]
```diff
@@ -458,4 +459,38 @@ object Utilities {
...
+  ): Unit = {
+    rules.foreach(rule => {
+      if (DatabaseDetailsCache.getDatabaseDetails(rule._2).isDefined) {
+        if (
+          databaseURLPriority(
+            DatabaseDetailsCache.getDatabaseDetails(rule._1).get.dbLocation,
+            DatabaseDetailsCache.getDatabaseDetails(rule._1).get.configFile
+          ) < databaseURLPriority(
+            dbUrl.value,
+            dbUrl.sourceFileOut.head.name
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/utility/Utilities.scala:478]: better to use `dbUrl.file.name.headOption.getOrElse("") or `"<unknown>"``

File: src/main/scala/ai/privado/utility/Utilities.scala [hunk 4]
```diff
@@ -458,4 +459,38 @@ object Utilities {
...
+          ) // Compare the priority of the database url with already present url in the database cache
+        ) {
+
+          DatabaseDetailsCache.removeDatabaseDetails(rule._2)
+          DatabaseDetailsCache.addDatabaseDetails(
+            DatabaseDetails(dbName, dbVendor, dbLocation, rule._1, dbUrl.sourceFileOut.head.name),
+            rule._2
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/utility/Utilities.scala:485]: similar comment here for file query

File: src/main/scala/ai/privado/tagger/config/RubyDBConfigTagger.scala [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+import ai.privado.languageEngine.java.language.NodeStarters
+import ai.privado.model.DatabaseDetails
+import ai.privado.tagger.{PrivadoParallelCpgPass, PrivadoSimpleCpgPass}
+import ai.privado.utility.Utilities.addDatabaseDetailsMultiple
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.JavaProperty
+import org.slf4j.LoggerFactory
+import overflowdb.traversal._
+
+class RubyDBConfigTagger(cpg: Cpg) extends PrivadoSimpleCpgPass(cpg) {
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/tagger/config/RubyDBConfigTagger.scala:13]: Can we not have a `BaseeDBConfigTagger` which have functions defined for parsing different database files like postgresDb dynamo db. And in the language specific tagger, we see any different implementation just override them?

Comment (khemrajrathore) [reply src/main/scala/ai/privado/tagger/config/RubyDBConfigTagger.scala:13]: Also this is a ruby tagger so it should be in language Engine ruby's package

Comment (khemrajrathore) [reply src/main/scala/ai/privado/tagger/config/RubyDBConfigTagger.scala:13]: Yes Please, we need to start modelling using the OOPS concept. I see we are writing a lot of duplicate code

Comment (ankit-privado) [reply src/main/scala/ai/privado/tagger/config/RubyDBConfigTagger.scala]: done

**Review by karan-batavia — APPROVED** [review]
LGTM

**Comment by khemrajrathore** [comment]
@ankit-privado Resolve the conflict and change the base to `addSchemaViolation` branch. We will get this merged

**Pull request merged**